lib/index.js

```diff
@@ -488,11 +482,29 @@ MapboxGeocoder.prototype = {
           this.eventManager.start(this);
           this.fresh = false;
         }
-        // supplement Mapbox Geocoding API results with locally populated results
-        res.features = res.features
-          ? localGeocoderRes.concat(res.features)
-          : localGeocoderRes;
 
+        if (this.options.localGeocoder) {
+
+          localGeocoderRes = this.options.localGeocoder(searchInput, res.features) || [];
+
+          // supplement Mapbox Geocoding API results with features returned by a promise
+          if(typeof localGeocoderRes.then === 'function') {
```

Comment:
the JSDoc would need updating if we go with localGeocoder, but as mentioned, I have a slight lean towards leaving localGeocoder as is and adding a new externalGeocoder option. Open to your thoughts.